website/content/specification/syntax/constraints.md

```diff
@@ -200,7 +200,9 @@ A `let` statement has a REQUIRED `@var` flag, which defines the variable name.
 
 A `let` statement has a REQUIRED `@expression` flag, which defines an [Metapath expression](/specification/syntax/metapath), whose result is used to define the variable's value in the evaluation context.
 
-During constraint evaluation, each `let` statement MUST be evaluated in encounter order. If a previous variable is bound with the same name in the evaluation context, the new value MUST bound in a sub-context to avoid side effects. This sub-context MUST be made available to any constraints following the `let` statement declaration, and to any constraints defined on child nodes of the current context.
+During constraint evaluation, each `let` statement MUST be evaluated in encountered order. If a previous variable is bound with the same name in the evaluation context, the new value MUST bound in a sub-context to avoid side effects. This sub-context MUST be made available to any constraints following the `let` statement declaration, and to any constraints defined on child nodes of the current context.
```

Comment:
I am not sure if this is blocking and/or material to this PR or not.

This is technically out of scope for #548, but given today's conversation, do want to address this concern within this PR and adjust given discussion and around processing order here and now as it has been decided with scope rebinding being disallowed or defer that? Note, the diff on this paragraph is the result of me correcting `encounter` to `encountered` to be more grammatical. I think this was the intent.

Reply:
Ouch, I had a comment here, somehow I dropped it.

Can we codify a rule that no two `let` element siblings have the same `name`?

Can we write a constraint for that?

@david-waltermire also grateful for your input here.